test/cmds/watch/watch.test.js

```diff
@@ -0,0 +1,148 @@
+const EventEmitter = require('events');
+const chokidar = require('chokidar');
+const watch = require('../../../dist/src/cmds/watch');
+const build = require('../../../dist/src/cmds/build/bundle');
+const fsUtils = require('../../../dist/src/utils/validate-fs');
+const miscUtils = require('../../../dist/src/utils/misc');
+
+describe('watch', () => {
+  describe('Watch a directory and its subdirectories for changes, and build when files are added or changed.', () => {
+
+    let watcherEmitter;
+
+    const mockArgv = {
+      src: 'index.js',
+      dist: 'dist',
+      outfileName: 'bundle.js',
+    };
+
+    beforeEach(() => {
+      jest.spyOn(chokidar, 'watch').mockImplementation(() => {
+        watcherEmitter = new EventEmitter();
+        watcherEmitter.add = () => undefined;
+        jest.spyOn(watcherEmitter, 'on');
+        jest.spyOn(watcherEmitter, 'add');
+        return watcherEmitter;
+      });
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+      jest.restoreAllMocks();
+      watcherEmitter = undefined;
+    });
+
+    it('successfully processes arguments from yargs', async () => {
+      jest.spyOn(console, 'log').mockImplementation();
+      const validateDirPathMock = jest.spyOn(fsUtils, 'validateDirPath').mockImplementation(() => true);
+      const validateFilePathMock = jest.spyOn(fsUtils, 'validateFilePath').mockImplementation(() => true);
+      const validateOutfileNameMock = jest.spyOn(fsUtils, 'validateOutfileName').mockImplementation(() => true);
+      const getOutfilePathMock = jest.spyOn(fsUtils, 'getOutfilePath').mockImplementation(() => 'dist/bundle.js');
+      await watch.handler(mockArgv);
+      expect(validateDirPathMock).toHaveBeenCalledTimes(1);
+      expect(validateFilePathMock).toHaveBeenCalledTimes(1);
+      expect(validateOutfileNameMock).toHaveBeenCalledTimes(1);
+      expect(getOutfilePathMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('watcher handles "changed" event correctly', async () => {
+      jest.spyOn(console, 'log').mockImplementation();
+      const bundleMock = jest.spyOn(build, 'bundle').mockImplementation();
+      const validateFilePathMock = jest.spyOn(fsUtils, 'validateFilePath').mockImplementation(() => true);
+
+      await watch.handler(mockArgv);
+      const finishPromise = new Promise((resolve, _) => {
+        watcherEmitter.on('change', () => {
+          expect(bundleMock).toHaveBeenCalledWith(mockArgv.src, `${mockArgv.dist}/${mockArgv.outfileName}`, mockArgv);
+          resolve();
+        });
+      });
+      watcherEmitter.emit('change');
+      await finishPromise;
+
+      expect(validateFilePathMock).toHaveBeenCalledTimes(1);
```

Comment:
I think we can remove the check for whether `validateFilePathMock` was called from this and all subsequent test cases.